include/mbedtls/mbedtls_config.h

```diff
@@ -1645,8 +1653,11 @@
  *
  * Enable TLS 1.3 ephemeral key exchange mode.
  *
- * Requires: MBEDTLS_ECDH_C, MBEDTLS_X509_CRT_PARSE_C, MBEDTLS_ECDSA_C or
- *           MBEDTLS_PKCS1_V21
+ * Requires: PSA_WANT_ALG_ECDH
```

Comment:
Shouldn't this be: `MBEDTLS_ECDH_C or (MBEDTLS_USE_PSA_CRYPTO and PSA_WANT_ALG_ECDH)`?

Reply:
Oh, I totally overlooked this! Even in `check_config.h` the requirement is for `MBEDTLS_PK_HAVE_ECDH` indeed, not just `PSA_WANT_ALG_ECDH` O.O

Reply:
Oh, I totally overlooked this! Even in check_config.h the requirement is for MBEDTLS_PK_HAVE_ECDH indeed, not just PSA_WANT_ALG_ECDH O.O

Reply:
For TLS 1.3, this should really be `PSA_WANT_ALG_ECDH`, as the TLS 1.3 always uses PSA for the ECDH computation, regardless of whether `MBEDTLS_USE_PSA_CRYPTO` is defined or not. (Evidence: `grep mbedtls_ecdh libary/ssl_tls13*.c` finds nothing, and this is not shared with TLS 1.2 - the only occurrences in `ssl_tls.c` are about 1.2 only. Same for `ECDH_C`.) So, thanks for noticing the discrepancy between the two files! I'm fixing `check_config.h`.

(Things are different for ECDSA as there TLS 1.3 does not call PSA functions directly, but uses PK, which dispatch to PSA or not based on `USE_PSA`. I know this is a mess and as usual I can't wait for the transition period to end with everything being all PSA all the time.)

Reply:
I'm glad @valeriosetti changed my mind about this being single-reviewer, thank you @mprse  for catching this!